src/aind_ephys_rig_qc/temporal_alignment.py

```diff
@@ -7,17 +7,240 @@
+    stream_folder_names = [
+        stream_folder_name.split("#")[-1]
+        for stream_folder_name in stream_folder_names
+    ]
+
+    figure, ax = plt.subplots(
+        2, len(lines_to_scan), figsize=(12, 5), layout="tight"
+    )
+
+    # check if distribution is uniform
+    # and plot distribution of inter-event intervals
+    # initialize p_value and p_short
+    p_value = np.zeros(len(lines_to_scan))
+    p_short = np.zeros(len(lines_to_scan))
+    bin_size = 100  # bin size in s to count number of events
+    for line_ind, curr_line in enumerate(lines_to_scan):
+        curr_events = events[
+            (events.stream_name == nidaq_stream_name)
+            & (events.processor_id == nidaq_stream_source_node_id)
+            & (events.state == 1)
+            & (events.line == curr_line)
+        ].copy()
+        bin_size = 100  # bin size in s
+        ts = curr_events.timestamp.values
+        bins = np.arange(np.min(ts), np.max(ts), bin_size)
+        bins_intervals = np.arange(0, 1.5, 0.1)
+        event_intervals = np.diff(ts)
+        ts = ts[np.where(event_intervals > 0.1)[0] + 1]
+        ax[0, line_ind].hist(event_intervals, bins=bins_intervals)
+        ax[0, line_ind].set_title(curr_line)
+        ax[0, line_ind].set_xlabel("Inter-event interval (s)")
+        ax[1, line_ind].hist(ts, bins=bins)
+        ax[1, line_ind].set_xlabel("Time in session (s)")
+
+        if line_ind == 0:
+            ax[0, line_ind].set_ylabel("Number of events")
+            ax[1, line_ind].set_ylabel("Number of events")
+
+        # check if distribution is uniform
+        ts_count, _ = np.histogram(ts, bins=bins)
+        expected_data = np.full(len(ts_count), np.mean(ts_count))
+        chi2_stat, p_value[line_ind] = chisquare(ts_count, expected_data)
+        # check if there's inter-event interval < 0.1s
+        p_short[line_ind] = np.sum(event_intervals < 0.05) / len(
+            event_intervals
+        )
+        if line_ind == 0:
+            ax[1, line_ind].set_title(
+                f"p_uniform time {p_value[line_ind]:.2f}"
+                + f"short interval perc {p_short[line_ind]:.2f}"
+            )
+        else:
+            ax[1, line_ind].set_title(
+                f"{p_value[line_ind]:.2f}, {p_short[line_ind]:.2f}"
+            )
+
+    # pick the line with even distribution overtime
+    # and has short inter-event interval
+    candidate_lines = lines_to_scan[(p_short > 0.5) & (p_value > 0.95)]
+    plt.suptitle(f"Harp line(s) {candidate_lines}")
+
+    if pdf is not None:
+        pdf.add_page()
+        pdf.set_y(30)
+        pdf.embed_figure(figure)
+    harp_line = candidate_lines
+
+    figure.savefig(os.path.join(directory, "harp_line_search.png"))
+    return harp_line, nidaq_stream_name, nidaq_stream_source_node_id
+
+
+def replace_original_timestamps(
+    directory,
+    original_timestamp_filename="original_timestamps.npy",
+    sync_timestamp_file="localsync_timestamps.npy",
```

Comment:
I would just call them `timestamps.npy`, so the open ephys reader will load them automatically downstream